packages/react/lib/components.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+/* global __webpack_modules__, __webpack_require__ */
+
+const { createElement, PureComponent } = require('react');
+const { default: withRouter } = require('react-router-dom/es/withRouter');
+
+exports.Miss = withRouter(({ staticContext }) => {
+  if (staticContext) {
+    staticContext.miss = true;
+  }
+  return null;
+});
+
+exports.Status = withRouter(({ staticContext, code }) => {
+  if (staticContext) {
+    staticContext.status = code;
+  }
+  return null;
+});
+
+exports.Header = withRouter(({ staticContext, name, value }) => {
+  if (staticContext) {
+    staticContext.headers = { ...staticContext.headers, [name]: value };
+  }
+  return null;
+});
+
+exports.Import = ({ loader, loading, weakId, module }) =>
+  withRouter(
+    class ImportComponent extends PureComponent {
+      constructor({ staticContext }) {
+        super();
+        if (__webpack_modules__[weakId] || staticContext) {
+          this.state = {
+            Component: ((value) => value.default || value)(
```

Comment:
This won't work with components that use named exports.

Reply:
Yeah, that is true. I would argue we can tackle this later iff we need to - either here or downstream. react-loadable does so by supporting and optional `render` function:

https://github.com/jamiebuilds/react-loadable/blob/master/README.md#customizing-rendering

Reply:
I have tackled this now by adding a second (optional) argument to `Import(module [, name])`. Of course, it defaults to `default` :grin: